python/src/scipp/plot/plot1d.py

```diff
@@ -0,0 +1,124 @@
+        sp.savefig(filename)
+
+    return sp
+
+
+class SciPlot1d(SciPlot):
+    def __init__(self,
+                 scipp_obj_dict=None,
+                 axes=None,
+                 errorbars=None,
+                 masks={"color": "k"},
+                 ax=None,
+                 pax=None,
+                 figsize=None,
+                 mpl_line_params=None,
+                 norm=None,
+                 vmin=None,
+                 vmax=None,
+                 scale=None,
+                 grid=False,
+                 title=None):
+
+        super().__init__(scipp_obj_dict=scipp_obj_dict,
+                         axes=axes,
+                         norm=norm,
+                         vmin=vmin,
+                         vmax=vmax,
+                         errorbars=errorbars,
+                         masks=masks,
+                         view_ndims=1)
+
+        self.widgets = PlotWidgets(axes=self.axes,
+                                   ndim=self.ndim,
+                                   name=self.name,
+                                   dim_to_shape=self.dim_to_shape,
+                                   masks=self.masks,
+                                   button_options=['x'])
+
+        # The model which takes care of all heavy calculations
+        self.model = PlotModel1d(scipp_obj_dict=scipp_obj_dict,
+                                 axes=self.axes,
+                                 name=self.name,
+                                 dim_to_shape=self.dim_to_shape,
+                                 dim_label_map=self.dim_label_map)
+
+        # The view which will display the 1d plot and send pick events back to
+        # the controller
+        self.view = PlotView1d(ax=ax,
+                               figsize=figsize,
+                               errorbars=self.errorbars,
+                               norm=norm,
+                               title=title,
+                               unit=self.params["values"][self.name]["unit"],
+                               masks=self.masks,
+                               mpl_line_params=mpl_line_params,
+                               picker=True,
+                               grid=grid)
+
+        # Profile view which displays an additional dimension as a 1d plot
+        if self.ndim > 1:
+            pad = config.plot.padding.copy()
+            pad[2] = 0.77
+            self.profile = ProfileView(
+                errorbars=self.errorbars,
+                ax=pax,
+                unit=self.params["values"][self.name]["unit"],
+                masks=self.masks,
+                figsize=(1.3 * config.plot.width / config.plot.dpi,
+                         0.6 * config.plot.height / config.plot.dpi),
+                padding=pad,
+                legend={
+                    "show": True,
+                    "loc": (1.02, 0.0)
+                })
+
+        # An additional panel view with widgets to save/remove lines
+        if self.ndim > 1:
```

Comment:
This is not performed inside the view (which would be `view1d`), it's the `SciPlot` constructor that creates all the pieces (model, view, widgets...) and sends them to the constructor of the `controller`.

The `panel` is just an extra set of widgets. Maybe you are saying they should be part of `widgets`?
Like a `widgets1d` that inherits from and adds to `widgets`?
i think that using some separate object instead of inheritance here might make things easier in the future if you want to add more widgets, swap some others out.

Reply:
I imagined that the between the model-view-controller, the behaviour would be self-directing. Maybe this will come out in the wash when this is all put under test.